src/query-builder/SoftDeleteQueryBuilder.ts

```diff
@@ -198,6 +198,21 @@ export class SoftDeleteQueryBuilder<Entity extends ObjectLiteral>
                 { type: "simple", condition: condition },
             ]
         if (parameters) this.setParameters(parameters)
+
+        // only update rows that are not already soft deleted
+        this.expressionMap.wheres.push({
```

Comment:
placing the fix in the "where" function doesn't adress 100% of the issue: one may not use the `where` function (i.e. soft deleting the whole table)

I would put the fix in `createUpdateExpression` instead

Reply:
This code adds the expression to the `where` list. even if the user doesn't pass any `where` clause, this would still be there when the query is executed because we hardcoded it inside the `where` list, or am I missing something?

Reply:
I think I get it, you mean if the user doesn't pass any `where` params (meaning this function never gets executed). I'll do the fix

Reply:
done 🙌